src/app/(proper_react)/redesign/(authenticated)/user/dashboard/page.tsx

```diff
@@ -33,7 +40,45 @@ export default async function DashboardPage() {
     return redirect("/redesign/user/welcome/");
   }
 
+  // This contains the latest scan results in our database.
   const latestScan = await getLatestOnerepScanResults(profileId);
+
+  // Attempt to fetch the current scan results from the provider.
+  // If anything is newer, add it to the database.
+  // Continue if there are any errors.
+  try {
+    const currentScan = await getAllScanResults(profileId);
+    const newScanResults: ScanResult[] = [];
+    currentScan.forEach((current) => {
+      latestScan.results.forEach((latest) => {
+        console.debug(new Date(current.updated_at), latest.updated_at);
+        if (
+          current.id === latest.onerep_scan_result_id &&
+          current.scan_id === latest.onerep_scan_id &&
+          new Date(current.updated_at) > latest.updated_at
+        ) {
+          newScanResults.push(current);
+        }
+      });
+    });
+
+    if (newScanResults.length > 0) {
+      const scanId = latestScan?.scan?.id;
+      if (!scanId && typeof scanId === "number") {
```

Comment:
I might be misunderstanding this, but doesn't this only trigger if `scanId === 0`? I'd expect:

```suggestion
      if (typeof scanId !== "number") {
```